### DISCUSSION TITLE
Vercel Deployment Problem

### DISCUSSION THREAD

**Discussion by janneserichsen** [neuefische/questions#469] [Web] [answered]
Hello i have an problem with deploying my project on vercel when i click redeploy this error comes 
<img width="331" alt="Bildschirmfoto 2023-01-25 um 16 12 57" src="https://user-images.githubusercontent.com/118190990/214600731-08824b19-6ce2-41a7-99ac-ff8582e97137.png">
<img width="1202" alt="Bildschirmfoto 2023-01-25 um 16 13 05" src="https://user-images.githubusercontent.com/118190990/214600748-06586bce-b2df-4133-9a10-8096a27b3741.png">
<img width="1196" alt="Bildschirmfoto 2023-01-25 um 16 13 11" src="https://user-images.githubusercontent.com/118190990/214600760-3609749f-179e-4721-8ccd-db584d512164.png">

**Comment by doemser** [comment, marked as answer]
This component is a page and that is problematic:

```javascript
export default function TaskList({ todos, onChangeTodo, onDeleteTodo }) {
  return (
    <ul>
      {todos.map((todo) => (
        <li key={todo.id}>
          <Task todo={todo} onChange={onChangeTodo} onDelete={onDeleteTodo} />
        </li>
      ))}
    </ul>
  );
}

```

You could use a page route as a component but I would not recommend doing it. In this case you are not using this component at all, just when you go to the page and then you do not have the props available you wish to get ( todos, onChangeTodo, onDeleteTodo )  and that is why the deployment is failing.

Keep in mind that folders in pages become routes through which you reach other pages. If you want to outsource your Tasklist as a component, create a components folder next to the pages folder and place it there.

**Reply by janneserichsen** [reply]
i´ve done this now this comes up 
Error: Command "npm run build" exited with 1

npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility

**Reply by doemser** [reply]
This warning should not actually cause the error, our current template has this warning and it is not a big deal. 

Please provide more details to your comments so that we can troubleshoot faster and better.

**Reply by janneserichsen** [reply]
i tried to redeploy and it doesnt work and in the error box this error comes up
Command "npm run build" exited with 1
<img width="968" alt="Bildschirmfoto 2023-01-26 um 16 48 55" src="https://user-images.githubusercontent.com/118190990/214882183-c7994446-3a5d-4740-a2fe-8d105cb3ab6c.png">

**Reply by NikRoe** [reply]
Hey Jannes,

we need more information than a screenshot of the error code to help you fix your deployment issue.

Please provide us with the following:

- a PR with the current changes on your branch feature/feedpage
- describe what you changed on that branch in order to fix the issue 
- send us a screenshot of the whole error message to check whether it remained the same

Thanks in advance!

**Reply by janneserichsen** [reply]
Hey Niklas here is the Pull Request to the feedpage
i changed the setup of the Page dynamical together with Coach Domenik
and this is the Errro Message
Command "npm run build" exited with 1

**Reply by janneserichsen** [reply]
https://github.com/janneserichsen/CapStone-Graffstar/pull/5/files

**Reply by NikRoe** [reply]
Thanks for the link to the PR.

When checking the error message, I get the information that the build failed because there are pages without a React Component as default export in two of your files.

The issue you are currently facing is related to the structure of your app.

Your components folder is part of your pages folder and this should **not** be the case.
Everything that lies within the pages folder in a next project is a page.

You need to move your components folder out of the pages folder.

Your folder structure should look roughly like this:
![Screenshot 2023-01-27 at 10 52 36](https://user-images.githubusercontent.com/98757391/215057675-45d60491-5ccb-4d9a-9645-edfd1f546288.png)

Please change your folder structure accordingly, this should fix the deployment error.

Hope this helps! :)

**Reply by janneserichsen** [reply]
Okay im gonna try this thank your for your time Niklas

**Comment by janneserichsen** [comment]
yes now it works thank you

**Reply by NikRoe** [reply]
Awesome 🙌  

I'll mark this issue as done ✅
